client/src/components/TodoApp/TodoApp.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from "react";
+import styles from "./TodoApp.module.css";
+import MyToast from "../MyToast/MyToast";
+import {
+  handleNewItem,
+  getAllTasks,
+  deleteAllTasks,
+} from "../../serverApi/itemClient";
+import ListItems from "../ListItems/ListItems";
+import UserInput from "../UserInput/UserInput";
+import Sort from "../Sort/Sort";
+import Button from "../Button/Button";
+import { checkIfInputIsNotValid } from "../../utils/utils";
+
+const TodoApp = () => {
+  const [data, setData] = useState([]);
+  const [showToast, setShowToast] = useState(false);
+  const [toastProps, setToastProps] = useState({
+    toastType: "",
+    message: "",
+  });
+
+  const displayToast = (toastType, message) => {
+    setShowToast(true);
+    setToastProps({
+      toastType,
+      message,
+    });
+  };
+  const killToast = () => {
+    setTimeout(() => {
+      setShowToast(false);
+    }, 5000);
+  };
+  const removeIdFromData = (id) => {
+    setData((prevData) => prevData.filter((item) => item.id !== id));
+  };
+  const checkIfTextAlreadyExist = (taskText) => {
+    const taskList = data.map((item) => item.itemName);
+    const result = taskList.includes(taskText) ? true : false;
+    return result;
+  };
+  const changeItemValueInData = (id, valueName, newValue) => {
+    setData((prevData) =>
+      prevData.map((item) => {
+        if (item.id === id) {
+          item[valueName] = newValue;
+        }
+        return item;
+      })
+    );
+  };
+
+  const NewInputHandler = async (userInputRef) => {
+    const userInput = userInputRef.current.value;
+    userInputRef.current.value = "";
+    if (checkIfInputIsNotValid(userInput)) {
+      displayToast("NEGATIVE", "Input is not valid");
+    } else {
+      const addNewItemsResult = await handleNewItem(userInput);
+      if (addNewItemsResult.result === "success") {
+        if (addNewItemsResult.data.length) {
+          setData([...data, ...addNewItemsResult.data]);
+          displayToast("POSITIVE", "Successfully add new items");
+
+          killToast();
+        } else if (addNewItemsResult.data.length === 0) {
+          displayToast("", "Try to add new tasks :)");
+        }
+      } else {
+        displayToast("NEGATIVE", addNewItemsResult.data);
```

Comment:
same comment about toast, this is way too coupled already.

if in the future you don't want a toast or you want to reuse this component you always have to take care you provide a toast handler, toast component etc.